src/Parser/ShadowPriest/Modules/Spells/Voidform.js

```diff
@@ -0,0 +1,255 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+
+
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Tab from 'Main/Tab';
+
+import { formatPercentage, formatNumber } from 'common/format';
+
+import Mindbender from './Mindbender';
+import Dispersion from './Dispersion';
+import VoidTorrent from './VoidTorrent';
+import VoidformsTab from './VoidformsTab';
+
+
+
+class Voidform extends Module {
+  static dependencies = {
+    dispersion: Dispersion,
+    voidTorrent: VoidTorrent,
+    mindbender: Mindbender,
+  };
+
+  _previousVoidformCast = null;
+  _totalHasteAcquiredOutsideVoidform = null;
+  _totalLingeringInsanityTimeOutsideVoidform = 0;
+  _voidformStatistics   = {};
+  _inVoidform           = false;
+  // _totalHasteAcquired = 0;
+
+
+  on_initialized() {
+    this.active = true;
+  }
+
+  get voidformStatistics() {
+    return Object.keys(this._voidformStatistics).map(key => this._voidformStatistics[key]);
+  }
+
+  get voidformAverageDuration(){
+    return this.voidformStatistics.reduce((p, c) => c.excluded ? p : p += c.stacks.length, 0) / this.voidformStatistics.reduce((p, c) => c.excluded ? p : p += 1, 0);
```

Comment:
Using (half)decent variable naming would make this code a lot easier to understand.